### ISSUE TITLE
[Feature request/Extension of current feature] HBD butteraugli RD tune addition to aomenc AVIF

### ISSUE THREAD

**Issue by BlueSwordM** [AOMediaCodec/libavif#899]
Currently, the butteraugli(from libjxl) RD tune is available in aomenc, but only for 8-bit inputs/outputs which is honestly a shame, since it currently is the best RD tune when it comes to raw low-high quality and fidelity.

It would be therefore be very nice if it worked consistently with 10b/12b outputs since this RD tune is extremely strong overall psycho-visually speaking, and is has a good quality/compute time cost overall.

Sadly, progress towards that seems to have stopped entirely in the related aomedia Gerrit issue, which is why I'm asking here:
https://bugs.chromium.org/p/aomedia/issues/detail?id=3129

If it would be possible to make it work for HBD inputs, it would not only be great for AVIF in general, but also be beneficial for video/animation AVIF/AV1.

**Comment by BlueSwordM** [comment]
@wantehchang If you have the time, perhaps you could look into it and give some insights for your team.
The butteraugli RD tune is top notch, and having it work on all bit depths would be awesome.

On another note, I discovered maybe why it didn't work properly in my patch: bad translation of YCbCr to RGB values, although there could always be something else causing the bugs.

**Comment by rathann** [comment]
Unfortunately, jpeg-xl has removed the API in v0.9.0: https://github.com/libjxl/libjxl/commit/a96301672a1b27a387c8f4906801b30ff64868a4 (https://github.com/libjxl/libjxl/pull/2576).

**Comment by jorismak** [comment]
Yeah so you need to use an older version, because the only thing this
integration picks up was the Butteraugli stuff.

And as mentioned in the libjxl ticket, that project is supposed to be a
sort of reference 'standard' implementation, and including butteraugli in
it means everybody would be forced to include a butteraugli implementation
if you wanted to be a libjxl alternative.
The plan is (was?) to move the metrics stuff to a separate library. Which
makes more sense, that projects like this can just use the metrics library
to get scores instead of having to include the encoder library for a
completely different image format, just for some metrics.

I don't know honestly if they ever got around to that. For the jxl team,
it's low priority.

And to be honest, with the 8 bit limitation and the results I ended up not
really bothering anymore for recent builds I use for my personal use.
